### PR TITLE
fix test case, examples and support more event source

### LINKED ISSUE #9975: Security Center Automation and Continuous Export is not working as expected
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
* `Terraform v0.13.4`
*  `azurerm v2.40.0`

### Affected Resource(s)

* `azurerm_security_center_automation`

### Terraform Configuration Files

```hcl
provider "azurerm" {
    features {}
}

data "azurerm_client_config" "current" {}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "westeurope"
}

resource "azurerm_eventhub_namespace" "example" {
  name                = "example-namespace123"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 2
}

resource "azurerm_security_center_automation" "example" {
  name                = "example-automation"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  action {
    type              = "EventHub"
    resource_id       = azurerm_eventhub_namespace.example.id
    connection_string = azurerm_eventhub_namespace.example.default_primary_connection_string
  }

  source {
    event_source = "Alerts"
    rule_set {
      rule {
        property_path  = "properties.metadata.severity"
        operator       = "Equals"
        expected_value = "High"
        property_type  = "String"
      }
    }
  }

  scopes = ["/subscriptions/${data.azurerm_client_config.current.subscription_id}"]
}
```

### Debug Output

https://gist.github.com/benhurjoel/57bad9b2b5cb7410a1bb5005389962e0

### Expected Behaviour

Continuous export should be visible on the Azure Portal post deployment. Tried both with EventHub and LogAnalytics type  

### Actual Behaviour

I notice that Eventhub name and Auth Roles names are required to set up continuous export (For EventHub type).  I tried setting up Continuous Export on Azure Portal manually and I have to choose the following attributes  :

image

But, I do not see a way to specify them using the Terraform resource. Only EventHub Namespace is available to specify.

The available options using Terraform : 
```hcl
  action {
    type              = "EventHub"
    resource_id       = azurerm_eventhub_namespace.example.id
    connection_string = azurerm_eventhub_namespace.example.default_primary_connection_string
  }
```

### Steps to Reproduce

1. `terraform apply`

### References

https://docs.microsoft.com/en-us/azure/security-center/continuous-export?tabs=azure-portal

### PR DESCRIPTION
**Pull request by njuCZ**
fix #9975

doc: https://docs.microsoft.com/en-us/azure/security-center/continuous-export?tabs=azure-portal

from the portal, data export to eventhub resource depends on eventhub, and eventhub authorization rule. The original example is not right, though the original acctest could success...

image

### CODE REVIEW

File: azurerm/internal/services/securitycenter/security_center_automation_resource.go [hunk 1]
```diff
@@ -128,9 +128,11 @@ func resourceSecurityCenterAutomation() *schema.Resource {
...
 							Required: true,
 							ValidateFunc: validation.StringInSlice([]string{
-								"Alerts",
-								"Assessments",
-								"SubAssessments",
+								string(security.Alerts),
+								string(security.Assessments),
+								string(security.SecureScoreControls),
+								string(security.SecureScores),
+								string(security.SubAssessments),
```
Comment (tombuildsstuff) [inline azurerm/internal/services/securitycenter/security_center_automation_resource.go:135]: can we update the documentation to include these new values? this should also likely be case-sensitive (by changing the `true` below)

**Review by tombuildsstuff — COMMENTED** [review]
hey @njuCZ 

Thanks for this PR - taking a look through if we can add these new values to the documentation then this otherwise LGTM 👍 

Thanks!

File: azurerm/internal/services/securitycenter/security_center_automation_resource.go (hunk 1 above)
Comment (njuCZ) [reply azurerm/internal/services/securitycenter/security_center_automation_resource.go:135]: sorry for missing. Have done it!

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Comment by tombuildsstuff** [comment]
Tests look ok 👍

**Pull request merged**